A man page was created (#142)
AntelopeIO/DUNE: docs/man/dune.1.md

---

> title: dune

Reviewer: ```suggestion
title: DUNE
```

See https://man7.org/linux/man-pages/man7/man-pages.7.html for best practices.

section: 1

> header: User Manual

Reviewer: ```suggestion
header: Docker Utilities for Node Execution
```

> footer: dune 1.1.0

Reviewer: ```suggestion
footer: AntelopeIO
```
The following three options all seem acceptable to me:
- Organization: "AntelopeIO"
- Version: "1.1.0" or "v1.1.0"
- name + ver: "DUNE 1.1.0"

@iamveritas What do you think? Personally, I like organization best. Best practices are unclear which is to be selected; however, a really quick review of existing man pages on my machine seems to indicate all of the above are acceptable.

Reviewer: Upon further consideration, we should avoid version information in the manpage unless we auto generate it. There is too much risk we will forget to update it and there is no requirement to include version.

Reviewer: With that, I recommend we use the organization: `footer: AntelopeIO`.

**`--remove-app <PROJ_DIR> <APP_NAME>`**

    Remove an application from the specified smart contract project.

**`--remove-lib <PROJ_DIR> <LIB_NAME>`**

    Remove a library from the specified smart contract project.

**`--remove-dep <PROJ_DIR> <OBJ_NAME> <DEP_NAME>`**

    Remove a dependency from the specified smart contract project.

**`--update-app <PROJ_DIR> <APP_NAME> <LANG> [CMPLR_OPTS] [LINK_OPTS]`**

    Update an application in the given smart contract project.

**`--update-lib <PROJ_DIR> <LIB_NAME> <LANG> [CMPLR_OPTS] [LINK_OPTS]`**
 
    Update a library in the given smart contract project.

**`--update-dep <PROJ_DIR> <OBJ_NAME> <DEP_NAME> [LOCATION] [TAG/RELEASE] [HASH]`**

    Update a dependency in the given smart contract project.
 
**`--build-project <PROJ_DIR>`**

    Build the given smart contract project.
                        
**`--clean-build-project <PROJ_DIR>`**

    Clean the given project and rebuild it from scratch.
                        
**`--validate <PROJ_DIR>`**

    Validate the given smart contract project.
                        
**`--populate <PROJ_DIR>`**

    Populate the given smart contract project.

> 

Reviewer: Missing `EXIT STATUS` section.

> # BUGS

Reviewer: ```suggestion
# REPORTING BUGS
```

Per best practices, "bugs" is where WE report known defects and "reporting bugs" is an acceptable section to describe where and how to submit bugs.

Please submit bug reports online at https://github.com/AntelopeIO/DUNE/issues

> # SEE ALSO

Reviewer: ```suggestion
# SEE ALSO

  **cdt**(1), **leap**(1)
  
#  COLOPHON
```

I think this might be better?

Reviewer: Or something like this? "See also" would normally describe other man pages. I'm not 100% sure here if we should have the hyperlink here or in "colophone"? Maybe @iamveritas has an opinion?

> For more details consult the full documentation and sources https://github.com/AntelopeIO/DUNE

Reviewer: We might want to add more text here? IDK.  Something like:

```
This page is part of version 1.1.0 of Docker Utilities for Node Execution. A description of the project, information 
about reporting bugs, and the latest version of this page can be found at https://github.com/AntelopeIO/DUNE/.
```
Again, @iamveritas 's input is appreciated.

Reviewer 2: the version is already listed on the top. to me it is enough as it is.

Reviewer 2: the ALL caps titles are a burden for my eyes though :)
is this style necessary for the man standard?

Reviewer: @iamveritas It would look bad if we were out of sync with the standard, I believe. To be honest, this is simply a header, though, and most users probably won't notice it. They will look at the body of the page.

Author: @iamveritas We agreed with @ScottBailey  to remove all information about the version at all, because, as I understand, the version is generating automatically and we risk to break version consistency.

Here is a manual how to write man pages: https://man7.org/linux/man-pages/man7/man-pages.7.html

Requirement for the title from this document:

       title  The title of the man page, written in all caps (e.g., MAN-
              PAGES).

---
title: DUNE
section: 1
header: User Manual
footer: DUNE 0.0.1
date: April 04, 2023
---

> # NAME

Reviewer 2: In general the EOS/ENF docs do not use ALL caps for titles. Instead all main words in titles use capital letters except the ones that connect the main words, e.g. 'a', 'an', 'the', 'of', etc.

Author: DUNE is an abbreviation so I wrote it this way. Corrected.

Reviewer: > In general the EOS/ENF docs do not use ALL caps for titles.

Does DUNE fit the general case? I think all caps for the project name is unfortunate, but that's where we are.

One option for correcting this is changing the name. I hope we can address this soon. See #141

Reviewer: Furthermore, after a bit of investigation, [best practices](https://man7.org/linux/man-pages/man7/man-pages.7.html) indicate the title should be all caps.

Reviewer: Meets best practices now. TY

> # NAME

Reviewer 2: also each doc has one title, that starts with one hash # char and then the sub-titles follow and start with two or more hash chars, e.g. ## sub-title 1, #### sub-title N

if `unix man` integration can not be done respecting our current docs general guidelines I guess it is alright for this doc to be merged like this.

Author: Yes terminals don't supports different sizes of fonts so it is the only way to write a name of section which will be correctly converted to a format which man utility understand.

Reviewer: > in general an ENF/EOS markdown doc has one title,

@iamveritas This is the source file for generating a man page and is not intended for "direct" consumption. It's probably more appropriate to look at the result (`dune.1`) using a man viewer.

DUNE - Docker Utilities for Node Execution.

# SYNOPSIS

`dune <option> [<arguments>] [-- <commands>]`

# DESCRIPTION

**DUNE** is a utility to interact with the EOS blockchain system installed in a Docker container. 

> The main purpose of this utility is to provide all features needed for management of EOS in one

Reviewer 2: for management of EOS `blockchain`

Author: I hope this description is better. I took it from the readme.md

program and make this process simple for non experienced users. DUNE offers functionality to manage
smart contract development, deployment, and interaction with the blockchain. 

# OPTIONS

**`-- <COMMANDS>`** 

>     Adding of double dash and space at the end of the command line allows to call any commands in the container 
>     or append additional options to the end of a current call.

Reviewer 3: ```suggestion
    Adding a double dash and space at the end of the command line allows for calling any commands in the container
    or appending additional options to the end of a current call.
```

Reviewer 2: in general ENF/EOS docs guidelines avoid gerunds.

    Example: dune -- cleos --help

**`-h, --help`** 

    Print the help message and exit

**`-s, --start <NODE>`** 

    Start a new node with a given name

**`-c <CONFIG_DIR>, --config <CONFIG_DIR>`**

    Optionally used with --start, a path containing the config.ini file to use.

**`--stop <NODE>`**

    Stop a node with a given name.

**`--remove <NODE>`**

    Remove a node with a given name. If the node is running it will be stopped.

**`--list`**

    Print list of all available nodes and their status.

**`--simple-list`**

    Print the same as --list but without formatting.

**`--set-active <NODE>`**

    Make the given node active.

**`--get-active`**

    Print name of the active node.

**`--export-node <NODE> <PATH>`**

    Export state and blocks log of the given node. PATH may be a directory or a filename with `.tgz` extension.
    
    Example: dune --export-node node ~/storage/ 

**`--import-node <NODE> <PATH>`**

    Import state and blocks log to a given node. PATH *must* be a path to a file which contains previously exported 

>     node with `.tgz` extention.

Reviewer 2: extention ->  extension

**`--system-newaccount <NAME> [CREATOR] [PUB_KEY] [PRIV_KEY] [-- OPTIONS]`**

    Create an EOSIO account with initial resources using "cleos system newaccount" command.
    This options calls cleos utility in the Docker container 
    so you may send additional cleos command line options to the call as OPTIONS

    Example: dune --system-newaccount myaccount -- --buy-ram-bytes 3000
    
**`--create-cmake-app <PROJ_NAME> <DIR>`**

    Create a new smart contract project at the given location.

**`--create-bare-app <PROJ_NAME> <DIR>`**

    Create a new empty smart contract project at the given location.
     
**`--cmake-build <DIR> [-- OPTIONS]`**

    Build a smart contract project at the given directory.
    Additional CMake options can be added to CMake call as OPTIONS 
     
    Example: dune --cmake-build ~/project -- -DFLAG1=On -DFLAG2=Off               

**`--ctest <DIR> [-- OPTIONS]`**

    Run the ctest tests for a smart contract project at the directory given.
    Additional ctest options can be added to ctest call as OPTIONS
     
    Example: dune --ctest ~/project -- --progress -V

**`--gdb <PROGRAM> [-- OPTIONS]`**

>     Start gdb into the container with given executive binary

Reviewer 3: ```suggestion
    Start gdb in the container with given executive binary
```

Author: When I wrote "into" @ScottBailey changed it to "in". He is a native English speaker so I believe he knows better :)

>     Additional gdp options can be added to the call as OPTIONS               

Reviewer 3: ```suggestion
    Additional gdb options can be added to the call as OPTIONS               
```

**`--deploy <DIR> <ACCOUNT>`**

    Deploy a smart contract to a given account.

**`--destroy-container`**

    Destroy the current Docker container. Warning! This will destroy your state and block log.

**`--stop-container`**

    Stop the current Docker container.

**`--start-container`**

    Start the current Docker container.

**`--set-core-contract <ACCOUNT>`**

    Set the core contract to the specified account (use `eosio` as account for normal system setup).

**`--set-bios-contract <ACCOUNT>`**

    Set the bios contract to the specified account (use `eosio` as account for normal system setup).

**`--set-token-contract <ACCOUNT>`**

    Set the token contract to the specified account (use `eosio.token` as account for normal system setup).

**`--bootstrap-system`**

>     Do setup of typical configration settings to prepare EOS system for work

Reviewer 2: configration -> configuration

**`--remove-app <PROJ_DIR> <APP_NAME>`**

    Remove an application from the specified smart contract project.

**`--remove-lib <PROJ_DIR> <LIB_NAME>`**

    Remove a library from the specified smart contract project.

**`--remove-dep <PROJ_DIR> <OBJ_NAME> <DEP_NAME>`**

    Remove a dependency from the specified smart contract project.

**`--update-app <PROJ_DIR> <APP_NAME> <LANG> [CMPLR_OPTS] [LINK_OPTS]`**

    Update an application in the given smart contract project.

**`--update-lib <PROJ_DIR> <LIB_NAME> <LANG> [CMPLR_OPTS] [LINK_OPTS]`**
 
    Update a library in the given smart contract project.

**`--update-dep <PROJ_DIR> <OBJ_NAME> <DEP_NAME> [LOCATION] [TAG/RELEASE] [HASH]`**

    Update a dependency in the given smart contract project.
 
**`--build-project <PROJ_DIR>`**

    Build the given smart contract project.
                        
**`--clean-build-project <PROJ_DIR>`**

    Clean the given project and rebuild it from scratch.
                        
**`--validate <PROJ_DIR>`**

    Validate the given smart contract project.
                        
**`--populate <PROJ_DIR>`**

    Populate the given smart contract project.

> # EXAMPLES

Reviewer 2: empty sections? pls either add content or remove the title

Author: I removed them but may be it make sense to write something in #AUTHORS ? Company/community name?

# AUTHORS

# BUGS

> Please submit bug reports online at: <https://github.com/AntelopeIO/DUNE/issues>

Reviewer 2: in general the EOS markdown files don't decorate links with angled braces.
either use the [link description](link url) or just link url without any decorations.

Author: [link description](link url) will not work so I removed the braces.

# SEE ALSO

> Full documentation and sources at: <https://github.com/AntelopeIO/DUNE>

Reviewer 2: For more details consult the [full documentation](https://github.com/AntelopeIO/DUNE#readme) and [sources](https://github.com/AntelopeIO/DUNE).

Author: man utility doesn't support the hidden URLs :(
